### PR TITLE
[gmp/nettle] try fixing build issues

### PR DESCRIPTION
**Pull request by Neumann-A**
Another approach of build nettle/gmp which maybe fixes the build issues?

### CODE REVIEW

File: ports/gmp/portfile.cmake [hunk 1]
```diff
@@ -56,6 +56,7 @@ if(VCPKG_TARGET_IS_WINDOWS)
         RELEASE_CONFIGURATION ${CONFIGURATION_RELEASE}
         DEBUG_CONFIGURATION ${CONFIGURATION_DEBUG}
         SKIP_CLEAN
+        OPTIONS /p:UseEnv=True
```
Comment (BillyONeal) [inline ports/gmp/portfile.cmake:59]: It seems like vcpkg_add_to_path should do this? We keep waffling back and forth on this setting.

Comment (Neumann-A) [reply ports/gmp/portfile.cmake:59]: The problem is that MSBuild simply ignores anything in the environment without that flag.

Comment (BillyONeal) [reply ports/gmp/portfile.cmake:59]: I see :(

File: ports/gmp/portfile.cmake [hunk 2]
```diff
@@ -11,8 +11,8 @@ if(VCPKG_TARGET_IS_WINDOWS)
     )
     vcpkg_find_acquire_program(YASM)
     get_filename_component(YASM_DIR "${YASM}" DIRECTORY)
-    vcpkg_add_to_path(${YASM_DIR})
-    set(ENV{YASMPATH} ${YASM_DIR}/)
+    vcpkg_add_to_path("${YASM_DIR}")
+    #set(ENV{YASMPATH} "${YASM_DIR}/")
```
Comment (JackBoosY) [inline ports/gmp/portfile.cmake:15]:
```suggestion
```
I think we should just remove this line.

File: ports/nettle/portfile.cmake [hunk 3]
```diff
@@ -12,8 +12,8 @@ if(VCPKG_TARGET_IS_WINDOWS)
     )
     vcpkg_find_acquire_program(YASM)
     get_filename_component(YASM_DIR "${YASM}" DIRECTORY)
-    vcpkg_add_to_path(${YASM_DIR})
-    set(ENV{YASMPATH} ${YASM_DIR}/)
+    vcpkg_add_to_path("${YASM_DIR}")
+    #set(ENV{YASMPATH} "${YASM_DIR}/")
```
Comment (JackBoosY) [inline ports/nettle/portfile.cmake:16]:
```suggestion
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
Related: #11502 #7733

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
The `nettle:x64-osx` regression happens again:
```
Undefined symbols for architecture x86_64:
  "_EVP_bf_ecb", referenced from:
      _openssl_bf128_set_encrypt_key in nettle-openssl.o
      _openssl_bf128_set_decrypt_key in nettle-openssl.o
  "_EVP_cast5_ecb", referenced from:
      _openssl_cast128_set_encrypt_key in nettle-openssl.o
      _openssl_cast128_set_decrypt_key in nettle-openssl.o
ld: symbol(s) not found for architecture x86_64
```

**Comment by wrobelda** [comment]
> The `nettle:x64-osx` regression happens again:
> 
> ```
> Undefined symbols for architecture x86_64:
>   "_EVP_bf_ecb", referenced from:
>       _openssl_bf128_set_encrypt_key in nettle-openssl.o
>       _openssl_bf128_set_decrypt_key in nettle-openssl.o
>   "_EVP_cast5_ecb", referenced from:
>       _openssl_cast128_set_encrypt_key in nettle-openssl.o
>       _openssl_cast128_set_decrypt_key in nettle-openssl.o
> ld: symbol(s) not found for architecture x86_64
> ```

Trying to build nettle on macOS and seeing this exact error. Should this be reported as a separate issue?

**Comment by JackBoosY** [comment]
@wrobelda Yes please.

**Comment by wrobelda** [comment]
> @wrobelda Yes please.

https://github.com/microsoft/vcpkg/issues/13949

**Pull request merged**